src/utils/details.js

```diff
@@ -84,9 +90,26 @@ function parseDetails ({
   secret
 }) {
   const detailsBuffer = Buffer.from(details, 'base64')
-  const publicRequest = _parseRequest(detailsBuffer)
-  const decrypted = cryptoHelper.aesDecryptBuffer(publicRequest.data, secret)
-  const privateRequest = _parseRequest(decrypted)
+  const publicRequest = _parseRequest({
+    request: detailsBuffer,
+    statusLine: true
+  })
+
+  const [ , token ] = publicRequest.headers['key'].match(KEY_HEADER_REGEX)
+  if (!token) {
+    throw new Error('invalid Key header in',
+      JSON.stringify(publicRequest.headers))
+  }
+
+  const paymentKey = cryptoHelper.getPaymentKey(
+    secret,
+    Buffer.from(token, 'base64'))
+
+  const decrypted = cryptoHelper.aesDecryptBuffer(publicRequest.data, paymentKey)
```

Comment:
Can you check to make sure the way this is done will work even with the largest possible data field?